File: maria/ruby/read/maria_read.rb
#! /usr/bin/ruby
# -*- coding: utf-8 -*-
#
#	maria_read.rb
#
#					Apr/07/2017
#
# ------------------------------------------------------------------
require 'mysql'
#
# ------------------------------------------------------------------
puts "*** 開始 ***"
#
host = "127.0.0.1"
user = "scott"
password = "tiger123"
data_base = 'city'
connection = Mysql::new(host, user,password,data_base)

sql_str = "SELECT id,name,population,date_mod FROM cities order by ID"
begin
	result = connection.query(sql_str)
	result.each do |row|
		print "#{row[0]}\t#{row[1]}\t#{row[2]}\t#{row[3]}\n"
	end
end
#
connection.close
#
puts "*** 終了 ***"
# ------------------------------------------------------------------

File: maria/node/read/maria_read.js
#! /usr/bin/node
// ---------------------------------------------------------------
//	maria_read.js
//
//					Jun/14/2020
//
// ---------------------------------------------------------------
'use strict'

var text_manipulate=require ("/var/www/data_base/common/node_common/text_manipulate")

// ---------------------------------------------------------------
console.error ("*** 開始 ***")

var mysql = require('mysql')

var connection = mysql.createConnection ({
	host: 'localhost',
	user: 'scott',
	password: 'tiger123'
	})

connection.query ('Use city')

connection.query("select * from cities", function (err, rows)
	{
  	if (err) throw err
  	console.log (rows.length)

	var dict_aa = new Object ()

	rows.forEach(function(row)
		{
		dict_aa[row.id] = {"name": row.name,
			"population": row.population,
			"date_mod": row.date_mod}
		})

	text_manipulate.dict_display_proc (dict_aa)

	connection.end()
	console.error ("*** 終了 ***")
	})

// ---------------------------------------------------------------
